packages/edit-site/src/components/style-book/index.js

```diff
@@ -346,33 +348,60 @@ function StyleBook( {
 /**
  * Style Book Preview component renders the stylebook without the Editor dependency.
  *
- * @param {Object}   props            Component props.
- * @param {string}   props.path       Path to the selected block.
- * @param {Object}   props.userConfig User configuration.
- * @param {Function} props.isSelected Function to check if a block is selected.
- * @param {Function} props.onSelect   Function to select a block.
+ * @param {Object}  props            Component props.
+ * @param {Object}  props.userConfig User configuration.
+ * @param {boolean} props.isStatic   Whether the stylebook is static or clickable.
  * @return {Object} Style Book Preview component.
  */
-export const StyleBookPreview = ( {
-	path = '',
-	userConfig = {},
-	isSelected,
-	onSelect,
-} ) => {
+export const StyleBookPreview = ( { userConfig = {}, isStatic = false } ) => {
 	const siteEditorSettings = useSelect(
 		( select ) => select( siteEditorStore ).getSettings(),
 		[]
 	);
 	// Update block editor settings because useMultipleOriginColorsAndGradients fetch colours from there.
 	dispatch( blockEditorStore ).updateSettings( siteEditorSettings );
 
+	const [ section, onChangeSection ] = useSection();
```

Comment:
Something about this hook placement triggers the following warning:


```
Cannot update a component (`BlockPreview`) while rendering a different component (`StyleBookPreview`). To locate the bad setState() call inside `StyleBookPreview`
```

For example, when navigating to an individual block's global style controls, and activating the style book.

It mean we might have to wrap the dispatch in a useEffect so the settings are updated after render:

```js
	useEffect( () => {
		dispatch( blockEditorStore ).updateSettings( siteEditorSettings );
	}, [ siteEditorSettings ] );
```

But I'm unsure of the side-effects - will it preserve the original intention to get the updated colors?

Reply:
Hmm that was merged in the classic stylebook PR though I'm not seeing the same error on trunk. I'll investigate.

Reply:
Ok so the error is caused by this PR rendering the `StyleBookPreview` component simultaneously with the global styles sidebar, and it seems to be complaining about the fact that we're updating the block editor settings inside `StyleBookPreview` while at the same time using those settings to render the `BlockPreview` inside the global styles sidebar. That's why the error only occurs when a specific block is selected in global styles and we click over to the stylebook view.

Wrapping the settings update in a `useEffect` does seem to fix the issue, so I'll add that in for now.

What I don't understand though is why we need to do this settings update in here at all. With the stylebook that renders inside the editor, the block editor settings are already up to date, but I can't work out whereabouts they're being updated. 

Another thing to consider is: should we be depending on the block editor settings here, or the site editor ones? Perhaps instead of using `useMultipleOriginColorsAndGradients` we should have an equivalent function that grabs the color palettes from the site editor settings?

Reply:
> What I don't understand though is why we need to do this settings update in here at all. 

I'm not sure either 🤔 I assumed it's because `useMultiOriginPalettes` needed the updated values [here](https://github.com/WordPress/gutenberg/blob/005a47e78c3733163271ea7f0f3e3f2369d41306/packages/edit-site/src/components/style-book/index.js#L132), but I commented the update line out and can't discern any difference when I update the palette in global styles and apply it to a block (the style book updates still).

Wasn't the update added as part of the class style book? https://github.com/WordPress/gutenberg/pull/66851

I am missing something, that much is clear. 

> Another thing to consider is: should we be depending on the block editor settings here, or the site editor ones?

I'd say site settings is right, because the [site editor is initialized](https://github.com/WordPress/gutenberg/blob/005a47e78c3733163271ea7f0f3e3f2369d41306/packages/edit-site/src/index.js#L36-L36) (and therefore the site editor store) with preloaded theme styles settings. 

> Perhaps instead of using useMultipleOriginColorsAndGradients we should have an equivalent function that grabs the color palettes from the site editor settings?

Conceptually, that sounds legit to me, but I haven't gone deeper than inspecting `useMultipleOriginColorsAndGradients` and `useSettings `. The context is always going to be global styles/theme json, right?

Reply:
> The context is always going to be global styles/theme json, right?

It goes a bit beyond theme.json because for classic themes it also picks up anything defined in the `editor-color-palette` and `editor-gradient-presets` theme supports. But that's when those palettes are in the block editor settings already.

I think it might be that the block editor settings only get updated when an actual block editor renders. In which case, if we're in the site preview, it makes sense that block editor settings aren't up to date, but site editor settings (which are essentially the same shape) already are.